src/Invoice/Calculator/FinancialYearCalculator.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Kimai time-tracking app.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Invoice\Calculator;
+
+use App\Configuration\SystemConfiguration;
+
+class FinancialYearCalculator
+{
+    private \DateTimeInterface|null $financialYearStart = null;
+    private \DateTimeInterface|null $financialYearEnd = null;
+
+    public function __construct(private SystemConfiguration $systemConfiguration)
```

Comment:
```suggestion
    public function __construct(private readonly SystemConfiguration $systemConfiguration)
```

Reply:
Never execute any call in the constructor.
As this is used in the ServiceContainer from Symfony, these ghost proxy objects will be replaced by their real implementation which could potentially happen on many or even every call.
The `__construct` should only be used to assign required objects

Reply:
Fixed